File: csharp/hamming/Hamming.cs
﻿using System;
using System.Linq;

public static class Hamming
{
    public static int Distance(string strand1, string strand2)
    {
        if (strand1.Length != strand2.Length) throw new ArgumentException("Strands have different length");

        return strand1.Zip(strand2).Count(pair => pair.First != pair.Second);
    }
}

File: fsharp/hamming/Hamming.fs
module Hamming

let distance (strand1: string) (strand2: string) = 
    if strand1.Length <> strand2.Length then
        None
    else    
        Seq.zip strand1 strand2
        |> Seq.filter (fun (n1, n2) -> n1 <> n2)
        |> Seq.length
        |> Some
